sklearn/discriminant_analysis.py

```diff
@@ -943,59 +971,116 @@ class QuadraticDiscriminantAnalysis(
         "priors": ["array-like", None],
         "reg_param": [Interval(Real, 0, 1, closed="both")],
         "store_covariance": ["boolean"],
         "tol": [Interval(Real, 0, None, closed="left")],
+        "covariance_estimator": [HasMethods("fit"), None],
     }
 
     def __init__(
-        self, *, priors=None, reg_param=0.0, store_covariance=False, tol=1.0e-4
+        self,
+        *,
+        solver="svd",
+        shrinkage=None,
+        priors=None,
+        reg_param=0.0,
+        store_covariance=False,
+        tol=1.0e-4,
+        covariance_estimator=None,
     ):
+        self.solver = solver
+        self.shrinkage = shrinkage
         self.priors = priors
         self.reg_param = reg_param
         self.store_covariance = store_covariance
         self.tol = tol
+        self.covariance_estimator = covariance_estimator
 
-    @_fit_context(prefer_skip_nested_validation=True)
-    def fit(self, X, y):
-        """Fit the model according to the given training data and parameters.
-
-        .. versionchanged:: 0.19
-            ``store_covariances`` has been moved to main constructor as
-            ``store_covariance``.
+    def _solve_eigen(self, X, y, shrinkage, covariance_estimator):
+        """Eigenvalue solver.
 
-        .. versionchanged:: 0.19
-            ``tol`` has been moved to main constructor.
+        The eigenvalue solver uses the eigen decomposition of the
+        class covariances to compute the rotation and scaling
+        matrices used for scoring new samples. This solver supports
+        use of any covariance estimator.
 
         Parameters
         ----------
         X : array-like of shape (n_samples, n_features)
-            Training vector, where `n_samples` is the number of samples and
-            `n_features` is the number of features.
+            Training data.
 
-        y : array-like of shape (n_samples,)
-            Target values (integers).
+        y : array-like of shape (n_samples,) or (n_samples, n_targets)
+            Target values.
 
-        Returns
-        -------
-        self : object
-            Fitted estimator.
+        shrinkage : 'auto', float or None
+            Shrinkage parameter, possible values:
+              - None: no shrinkage.
+              - 'auto': automatic shrinkage using the Ledoit-Wolf lemma.
+              - float between 0 and 1: fixed shrinkage constant.
+
+            Shrinkage parameter is ignored if  `covariance_estimator` is
+            not None
+
+        covariance_estimator : estimator, default=None
+            If not None, `covariance_estimator` is used to estimate
+            the covariance matrices instead of relying the empirical
+            covariance estimator (with potential shrinkage).
+            The object should have a fit method and a ``covariance_`` attribute
+            like the estimators in sklearn.covariance.
+            if None the shrinkage parameter drives the estimate.
```

Comment:
Nitpick: could you please rewrap docstring paragraphs (respectively multi-line exception messages) to use more of the available 80 (respectively 88) columns allowed by our formatting conventions?